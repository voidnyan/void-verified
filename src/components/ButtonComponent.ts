import {DOM} from "../utils/DOM";

export class ButtonComponent {
	element: HTMLButtonElement;
	constructor(text: string, onClick: (event: Event) => void, classes?: string) {
		this.element = DOM.create("button", "button" + (classes ? ` ${classes}` : ""), text);
		this.element.addEventListener("click", onClick);
	}

	setText(text: string) {
		this.element.replaceChildren(text);
	}
}
