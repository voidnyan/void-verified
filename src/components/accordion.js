import { DOM } from "../utils/DOM";

const Accordion = (head, body, isOpen) => {
	const bodyWrapper = DOM.create("div", "accordion-body", body);
	bodyWrapper.setAttribute("data-is-open", isOpen);
	head.addEventListener("click", () => {
		if (bodyWrapper.getAttribute("data-is-open")) {
			bodyWrapper.setAttribute("data-is-open", false);
		} else {
			bodyWrapper.setAttribute("data-is-open", true);
		}
	});

	const accordion = DOM.create("div", "accordion");
	accordion.append(head, bodyWrapper);
	return accordion;
};
