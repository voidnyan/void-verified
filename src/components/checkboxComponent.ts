import {DOM} from "../utils/DOM";

export class CheckboxComponent {
	element: HTMLInputElement;
	constructor(checked: boolean, onChange?: (checked: boolean) => void) {
		this.element = DOM.create("input", "checkbox");
		this.element.setAttribute("type", "checkbox");
		this.element.checked = checked;

		if (onChange !== undefined) {
			this.element.addEventListener("change", (event) => {
				const target = event.target as HTMLInputElement;
				onChange(target.checked)
			});
		}
	}

	getValue(): boolean {
		return this.element.checked;
	}

	setValue(checked: boolean): void {
		this.element.checked = checked;
	}
}
