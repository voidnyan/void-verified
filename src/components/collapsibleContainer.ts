import {DOM} from "../utils/DOM";
import {ChevronDownIcon, ChevronUpIcon} from "../assets/icons";
import {LocalStorageKeys} from "../assets/localStorageKeys";

export class CollapsibleContainer {
	element: HTMLDivElement;
	head: HTMLDivElement;
	body: HTMLDivElement;
	private bodyWrap: HTMLDivElement;
	private headWrap: HTMLDivElement;
	private collapseIconContainer: HTMLDivElement;
	private isCollapsed = false;
	private readonly emptyContent;
	private collapseId?: string;
	constructor(head?: any, body?: any, classes?: {head?: string, body?: string}, emptyContent?: any, collapseId?: string) {
		this.emptyContent = emptyContent;
		this.collapseId = collapseId;
		if (this.collapseId) {
			this.isCollapsed = CollapseStatus.getIsCollapsed(this.collapseId);
		}
		this.element = DOM.create("div", "collapsible-container");

		this.createHead(head);
		this.createBody(body);

		this.element.append(this.headWrap, this.bodyWrap);

		this.setClasses(classes);
	}

	private createHead(head?: any) {
		this.head = DOM.create("div", "collapsible-container-head");
		this.collapseIconContainer = DOM.create("div", "collapsible-container-icon", ChevronDownIcon());
		this.headWrap = DOM.create("div", "collapsible-container-head-wrap");
		this.headWrap.addEventListener("click", () => {
			this.handleCollapse(!this.isCollapsed);
		})
		if (head) {
			if (head) {
				if (Array.isArray(head)) {
					this.head.append(...head);
				} else {
					this.head.append(head);
				}
			}
		}
		this.headWrap.append(this.head, this.collapseIconContainer);
	}

	private createBody(body?: any) {
		this.body = DOM.create("div", "collapsible-container-body");
		this.bodyWrap = DOM.create("div", "collapsible-container-body-wrap");
		this.bodyWrap.append(this.body);
		this.onDomLoad(body);
		if (this.isCollapsed) {
			this.handleCollapse(this.isCollapsed);
		}
	}

	handleCollapse(collapse: boolean) {
		this.isCollapsed = collapse;
		if (this.isCollapsed) {
			this.bodyWrap.classList.add("void-collapsed");
			// this.collapseIconContainer.replaceChildren(ChevronUpIcon());
		} else {
			this.bodyWrap.classList.remove("void-collapsed");
			// this.collapseIconContainer.replaceChildren(ChevronDownIcon());
		}
		if (this.collapseId) {
			CollapseStatus.save(this.collapseId, this.isCollapsed);
		}
	}

	setContent(body?: any) {
		this.body.replaceChildren();
		if (body) {
			if (Array.isArray(body)) {
				this.body.append(...body);
			} else {
				this.body.append(body);
			}
		} else if (this.emptyContent) {
			this.body.append(this.emptyContent);
		}

		const rect = this.body.getBoundingClientRect();
		this.bodyWrap.style.setProperty("--max-height", `${rect.height}px`);
	}

	private onDomLoad(body?: any) {
		if (!body && !this.emptyContent) {
			return;
		}

		const observer = new MutationObserver((mutationsList, observer) => {
			mutationsList.forEach(mutation => {
				mutation.addedNodes.forEach(node => {
					if (node === this.body || node.contains(this.body)) {
						this.setContent(body);
						observer.disconnect();
					}
				});
			});
		});

		observer.observe(document.body, { childList: true, subtree: true });
	}


	private setClasses(classes?: {head?: string, body?: string}) {
		if (classes?.head) {
			DOM.transformClasses(this.head, classes.head)
		}
		if (classes?.body) {
			DOM.transformClasses(this.body, classes.body)
		}
	}
}

class CollapseStatus {
	static save(collapseId: string, collapsed: boolean) {
		let collapseItems = JSON.parse(localStorage.getItem(LocalStorageKeys.collapsedContainers)) ?? [];
		if (collapseItems.find(x => x.collapseId === collapseId)) {
			collapseItems = collapseItems.map(x => x.collapseId === collapseId ? {...x, collapsed} : x);
		} else {
			collapseItems.push({collapseId, collapsed});
		}
		localStorage.setItem(LocalStorageKeys.collapsedContainers, JSON.stringify(collapseItems));
	}

	static getIsCollapsed(collapseId: string) {
		const collapsedItems = JSON.parse(localStorage.getItem(LocalStorageKeys.collapsedContainers)) ?? [];
		const isCollapsed = collapsedItems.find(x => x.collapseId === collapseId)?.collapsed ?? false;
		return isCollapsed;
	}
}
