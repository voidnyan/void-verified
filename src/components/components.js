import {
	DoubleChevronLeftIcon,
	DoubleChevronRightIcon,
	EyeClosedIcon,
	EyeIcon,
	HeartIcon,
} from "../assets/icons";
import { DOM } from "../helpers/DOM";

export const ColorPicker = (value, onChange) => {
	const container = DOM.create("div", "color-picker-container");
	const colorPicker = DOM.create("input", "color-picker");
	colorPicker.setAttribute("type", "color");
	colorPicker.value = value;
	colorPicker.addEventListener("change", (event) => {
		onChange(event);
	});

	container.append(colorPicker);
	const inputField = DOM.create("input", "color-picker-input");
	inputField.value = value ?? "#";
	inputField.addEventListener("change", (event) => {
		onChange(event);
	});
	container.append(inputField);
	return container;
};

export const InputField = (value, onChange, classes) => {
	const inputField = DOM.create("input", transformClasses("input", classes));
	inputField.value = value;
	inputField.addEventListener("change", (event) => {
		onChange(event);
	});
	return inputField;
};

export const SecretField = (value, onChange) => {
	const secret = InputField(value, onChange);
	secret.setAttribute("type", "password");
	const eyeIcon = EyeIcon();
	const closedEyeIcon = EyeClosedIcon();

	const container = DOM.create("div", "action-container", secret);
	const iconButton = IconButton(eyeIcon, (event) => {
		if (event.target.firstChild === eyeIcon) {
			event.target.replaceChildren(closedEyeIcon);
			secret.setAttribute("type", "text");
			return;
		}
		event.target.replaceChildren(eyeIcon);
		secret.setAttribute("type", "password");
	});
	container.append(iconButton);
	return container;
};

export const ActionInputField = (value, onClick, icon) => {
	const inputField = InputField(value, () => {});

	const container = DOM.create("div", "action-container", inputField);
	const iconButton = IconButton(icon, (event) => {
		onClick(event, inputField);
	});
	container.append(iconButton);
	return container;
};

export const RangeField = (value, onChange, max, step = 1, min = 0, unit) => {
	const container = DOM.create("div", "range-container");
	const range = DOM.create("input", "range");
	const display = DOM.create("div", "range-display", `${value}${unit ?? ""}`);
	range.setAttribute("type", "range");
	range.addEventListener("change", (event) => {
		onChange(event);
	});
	range.addEventListener("input", (event) => {
		display.replaceChildren(`${event.target.value}${unit ?? ""}`);
	});
	range.setAttribute("max", max);
	range.setAttribute("min", min);
	range.setAttribute("step", step);
	range.setAttribute("value", value);
	container.append(range, display);
	return container;
};

export const Button = (text, onClick) => {
	const button = DOM.create("button", "button", text);
	button.addEventListener("click", (event) => {
		onClick(event);
	});
	return button;
};

export const IconButton = (text, onClick, classes) => {
	const button = DOM.create(
		"div",
		transformClasses("icon-button", classes),
		text
	);
	button.addEventListener("click", (event) => {
		onClick(event);
	});
	return button;
};

export const Note = (text) => {
	const note = DOM.create("div", "notice", text);
	return note;
};

export const Link = (text, href, target = "_blank", classes) => {
	const link = DOM.create("a", transformClasses("link", classes), text);
	link.setAttribute("href", href);
	link.setAttribute("target", target);
	return link;
};

export const TextArea = (text, onChange, classes) => {
	const textArea = DOM.create(
		"textarea",
		transformClasses("textarea", classes),
		text
	);
	textArea.addEventListener("change", (event) => {
		onChange(event);
	});

	return textArea;
};

export const Toast = (message, type) => {
	const toast = DOM.create("div", transformClasses("toast", type), message);
	return toast;
};

export const Select = (options) => {
	const container = DOM.create("div", "select");
	for (const option of options) {
		container.append(option);
	}
	return container;
};

export const Option = (value, selected, onClick) => {
	const option = DOM.create("div", "option", value);
	if (selected) {
		option.classList.add("active");
	}
	option.addEventListener("click", onClick);
	return option;
};

export const Label = (text, element) => {
	const container = DOM.create("div", "label-container");
	const label = DOM.create("label", "label-span", text);
	const id = Math.random();
	label.setAttribute("for", id);
	element.setAttribute("id", id);
	container.append(label, element);
	return container;
};

export const Table = (head, body) => {
	const table = DOM.create("table", "table", [head, body]);
	return table;
};

export const TableHead = (...headers) => {
	const headerCells = headers.map((header) => DOM.create("th", null, header));
	const headerRow = DOM.create("tr", null, headerCells);
	const head = DOM.create("thead", null, headerRow);
	return head;
};

export const TableBody = (rows) => {
	const tableBody = DOM.create("tbody", null, rows);
	return tableBody;
};

export const Checkbox = (checked, onChange, title, disabled = false) => {
	const checkbox = DOM.create("input", "checkbox");
	checkbox.setAttribute("type", "checkbox");
	checkbox.checked = checked;

	if (disabled) {
		checkbox.setAttribute("disabled", "");
	}

	checkbox.addEventListener("change", onChange);
	checkbox.title = title;
	return checkbox;
};

export const SettingLabel = (text, input) => {
	const container = DOM.create("div", "setting-label-container", input);
	const label = DOM.create("label", "setting-label", text);
	const id = Math.random();
	label.setAttribute("for", id);
	input.setAttribute("id", id);
	container.append(label);
	return container;
};

export const GifKeyboard = (header) => {
	const container = DOM.create("div", "gif-keyboard-container");
	container.append(header);
	const gifList = DOM.create("div", "gif-keyboard-list");
	const controls = DOM.create("div", "gif-keyboard-control-container");
	container.append(
		DOM.create("div", "gif-keyboard-list-container", [controls, gifList])
	);

	return container;
};

export const GifItem = (url, onClick, onLike, gifs) => {
	const container = DOM.create("div", "gif-keyboard-item");
	container.addEventListener("click", () => {
		onClick();
	});
	const img = DOM.create("img");
	img.setAttribute("src", url);
	container.append(GifContainer(img, onLike, gifs));
	return container;
};

export const GifContainer = (imgElement, onLike, gifs) => {
	const container = DOM.create("div", "gif-like-container");
	container.append(
		imgElement,
		IconButton(
			HeartIcon(),
			(event) => {
				event.stopPropagation();
				event.preventDefault();
				onLike(event);
				if (event.target.classList.contains("void-liked")) {
					event.target.classList.remove("void-liked");
				} else {
					event.target.classList.add("void-liked");
				}
			},
			`gif-like ${gifs.includes(imgElement.src) && "liked"}`
		)
	);
	return container;
};

export const Pagination = (currentPage, maxPage, onClick) => {
	const container = DOM.create("div", "pagination-container");

	if (maxPage < 1) {
		return container;
	}

	let displayedPages = [];
	if (maxPage >= 3) {
		container.append(
			IconButton(
				DoubleChevronLeftIcon(),
				() => {
					onClick(0);
				},
				`pagination-skip ${currentPage === 0 && "active"}`
			)
		);
		if (currentPage >= maxPage - 1) {
			displayedPages.push(maxPage - 2, maxPage - 1, maxPage);
		} else if (currentPage === 0) {
			displayedPages.push(currentPage, currentPage + 1, currentPage + 2);
		} else {
			displayedPages.push(currentPage - 1, currentPage, currentPage + 1);
		}
	} else {
		for (let i = 0; i <= maxPage; i++) {
			displayedPages.push(i);
		}
	}

	for (const page of displayedPages) {
		container.append(
			IconButton(
				page + 1,
				() => {
					onClick(page);
				},
				currentPage === page && "active"
			)
		);
	}

	if (maxPage >= 3) {
		container.append(
			IconButton(
				DoubleChevronRightIcon(),
				() => {
					onClick(maxPage);
				},
				`pagination-skip ${currentPage === maxPage && "active"}`
			)
		);
	}

	return container;
};

const transformClasses = (base, additional) => {
	let classes = base;
	if (additional) {
		classes += ` ${additional}`;
	}
	return classes;
};
