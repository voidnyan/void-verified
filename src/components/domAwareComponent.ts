export class DomAwareComponent {
	onDomLoad(target: Node, callback: () => void) {
		const observer = new MutationObserver((mutationsList, observer) => {
			for (const mutation of mutationsList) {
				for (const node of mutation.addedNodes) {
					if (node === target || node.contains(target)) {
						callback();
						observer.disconnect();
					}
				}
			}
		});

		observer.observe(document.body, { childList: true, subtree: true });
	}

	onDomUnload(target: Node, callback: () => void) {
		const observer = new MutationObserver((mutations) => {
			for (const mutation of mutations) {
				for (const removedNode of mutation.removedNodes) {
					if (removedNode === target || removedNode.contains(target)) {
						callback();
						observer.disconnect();
					}
				}
			}
		});

		observer.observe(document.body, { childList: true, subtree: true });
	}
}
