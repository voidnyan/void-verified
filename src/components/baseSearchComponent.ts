import {DOM} from "../utils/DOM";
import {DomAwareComponent} from "./domAwareComponent";

export class BaseSearchComponent extends DomAwareComponent {
	element: HTMLDivElement
	resultsContainer: HTMLDivElement;
	constructor() {
		super();
		this.element = DOM.create("div", "search-container");
		this.resultsContainer = DOM.create("div", "search-results");
		document.body.append(this.resultsContainer);
		this.onDomLoad(this.element, () => {
			this.attachScrollListener()
		});
		this.onDomUnload(this.element, () => {
			this.resultsContainer.remove();
		});
	}

	updateResultsContainerLocation() {
		const elementRect = this.element.getBoundingClientRect();
		this.resultsContainer.style.top = `${elementRect.bottom + window.scrollY}px`;
		this.resultsContainer.style.left = `${elementRect.left + window.scrollX}px`;
		this.resultsContainer.style.width = `${elementRect.width}px`;
	}

	private attachScrollListener() {
		const scrollContainer = this.findScrollableParent(this.element) || window;
		scrollContainer.addEventListener("scroll", () => {
			this.updateResultsContainerLocation();
		}, { passive: true });
	}

	private findScrollableParent(el: HTMLElement): HTMLElement | null {
		let parent: HTMLElement | null = el.parentElement;
		while (parent) {
			const style = getComputedStyle(parent);
			const overflowY = style.overflowY;
			if (overflowY === "auto" || overflowY === "scroll") {
				return parent;
			}
			parent = parent.parentElement;
		}
		return null;
	}
}
