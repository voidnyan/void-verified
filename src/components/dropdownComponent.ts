import {DomAwareComponent} from "./domAwareComponent";
import {DOM} from "../utils/DOM";

interface IDropdownMenuOption {
	value: string,
	item: Node
}

export class DropdownMenuComponent extends DomAwareComponent {
	trigger: Element;
	menu: HTMLDivElement;
	constructor(options: string[] | IDropdownMenuOption[], trigger: Element, callback: (value: string) => void, initialValue?: string) {
		super();
		this.menu = DOM.create("div", "dropdown-menu");
		this.trigger = trigger;
		this.createMenu(options, callback, initialValue)
		this.trigger.addEventListener("click", () => {
			this.open();
		});

		this.onDomLoad(trigger, () => {
			document.body.append(this.menu);
		});

		this.onDomUnload(trigger, () => {
			this.menu.remove();
		});
	}

	private open() {
		this.menu.classList.add("void-visible");
		const triggerRect = this.trigger.getBoundingClientRect();
		const menuRect = this.menu.getBoundingClientRect();
		const padding = 5;
		this.menu.style.top = `${triggerRect.bottom + window.scrollY + padding}px`;
		this.menu.style.left = `${triggerRect.left + window.scrollX - menuRect.width + triggerRect.width}px`;

		const closeCall = (event) => {
			const menu = event.target.closest(".void-dropdown-menu");
			const t = event.target.closest([...this.trigger.classList].map(x => "." + x));
			if (menu !== this.menu && event.target !== this.trigger && t !== this.trigger) {
				this.close();
				document.removeEventListener("click", closeCall);
			}
		};

		document.addEventListener("click", closeCall);
	}

	private close() {
		this.menu.classList.remove("void-visible");
	}

	private createMenu(options: string[] | IDropdownMenuOption[], callback: (value: string | IDropdownMenuOption) => void, initialValue?: string) {
		for (const option of options) {
			let item;
			if (typeof option === "string") {
				item = DOM.create("div", "dropdown-menu-item");
				item.append(option);
				if (option === initialValue) {
					item.classList.add("void-active");
				}
			} else {
				item = option.item;
				item.classList.add("void-dropdown-menu-item");
				if (option.value === initialValue) {
					item.classList.add("void-active");
				}
			}
			item.addEventListener("click", () => {
				if (initialValue) {
					this.menu.querySelector(".void-active")?.classList.remove("void-active");
					item.classList.add("void-active");
				}
				callback(option);
				this.close();
			});
			this.menu.append(item);
		}
	}
}
