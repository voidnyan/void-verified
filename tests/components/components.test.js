import * as components from "../../src/components/components";

beforeAll(() => {
	jest.spyOn(global.Math, "random").mockReturnValue(0.12352);
});

afterAll(() => {
	jest.spyOn(global.Math, "random").mockRestore();
});

const componentBlackList = [
	"GifContainer",
	"GifItem",
	"Label",
	"Select",
	"SettingLabel",
];

const simpleComponents = Object.entries(components)
	.map(([name, component]) => {
		return { name, component };
	})
	.filter((component) => !componentBlackList.includes(component.name))
	.map((component) => {
		return [component.name, component.component];
	});

describe("Components", () => {
	test.each([...simpleComponents])(
		"%p matches snapshot",
		(ree, component) => {
			expect(component()).toMatchSnapshot();
		}
	);
});
