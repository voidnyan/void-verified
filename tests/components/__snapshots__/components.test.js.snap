// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components "ActionInputField" matches snapshot 1`] = `
<div
  class="void-action-container"
>
  <input
    class="void-input"
  />
  <div
    class="void-icon-button"
  />
</div>
`;

exports[`Components "Button" matches snapshot 1`] = `
<button
  class="void-button"
/>
`;

exports[`Components "Checkbox" matches snapshot 1`] = `
<input
  class="void-checkbox"
  title="undefined"
  type="checkbox"
/>
`;

exports[`Components "ColorPicker" matches snapshot 1`] = `
<div
  class="void-color-picker-container"
>
  <input
    class="void-color-picker"
    type="color"
  />
  <input
    class="void-color-picker-input"
  />
</div>
`;

exports[`Components "GifKeyboard" matches snapshot 1`] = `
<div
  class="void-gif-keyboard-container"
>
  undefined
  <div
    class="void-gif-keyboard-list-container"
  >
    <div
      class="void-gif-keyboard-control-container"
    />
    <div
      class="void-gif-keyboard-list"
    />
  </div>
</div>
`;

exports[`Components "IconButton" matches snapshot 1`] = `
<div
  class="void-icon-button"
/>
`;

exports[`Components "InputField" matches snapshot 1`] = `
<input
  class="void-input"
/>
`;

exports[`Components "Link" matches snapshot 1`] = `
<a
  class="void-link"
  href="undefined"
  target="_blank"
/>
`;

exports[`Components "Note" matches snapshot 1`] = `
<div
  class="void-notice"
/>
`;

exports[`Components "Option" matches snapshot 1`] = `
<div
  class="void-option"
/>
`;

exports[`Components "Pagination" matches snapshot 1`] = `
<div
  class="void-pagination-container"
/>
`;

exports[`Components "RangeField" matches snapshot 1`] = `
<div
  class="void-range-container"
>
  <input
    class="void-range"
    max="undefined"
    min="0"
    step="1"
    type="range"
    value="undefined"
  />
  <div
    class="void-range-display"
  >
    undefined
  </div>
</div>
`;

exports[`Components "SecretField" matches snapshot 1`] = `
<div
  class="void-action-container"
>
  <input
    class="void-input"
    type="password"
  />
  <div
    class="void-icon-button"
  >
    <svg
      class="w-6 h-6"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      
                
      <path
        d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      
                
      <path
        d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
      
            
    </svg>
  </div>
</div>
`;

exports[`Components "Table" matches snapshot 1`] = `
<table
  class="void-table"
>
  undefined
  undefined
</table>
`;

exports[`Components "TableBody" matches snapshot 1`] = `<tbody />`;

exports[`Components "TableHead" matches snapshot 1`] = `
<thead>
  <tr />
</thead>
`;

exports[`Components "TextArea" matches snapshot 1`] = `
<textarea
  class="void-textarea"
/>
`;

exports[`Components "Toast" matches snapshot 1`] = `
<div
  class="void-toast"
/>
`;
